### PR TITLE
sonar 4

### PR DESCRIPTION
**Pull request by a1dutch**

### CODE REVIEW

File: sonar4.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require "formula"
```
Comment (DomT4) [inline sonar4.rb]: Cut this line :)

File: sonar4.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <array>
+        <string>#{opt_bin}/sonar</string>
+        <string>start</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+    </dict>
+    </plist>
+    EOS
+  end
```
Comment (DomT4) [inline sonar4.rb]: A test down here would be great if possible. Simple is fine if necessary.

Comment (a1dutch) [inline sonar4.rb]: i tried adding a test but keep getting some 'block in system' error, the test below is what i was using

  test do
    system "#{bin}/sonar status"
  end

Comment (DomT4) [inline sonar4.rb]: That's odd. Does the same thing happen with:

```ruby
test do
  system bin/"sonar", "status"
end
```

?

Can you provide the whole block in system error given? Thanks!

Comment (a1dutch) [inline sonar4.rb]: here you go, same error as before

/usr/local/Library/Taps/homebrew/homebrew-versions: brew test sonar4
Testing sonar4
==> /usr/local/Cellar/sonar4/4.5.4/bin/sonar status

/usr/local/Cellar/sonar4/4.5.4/bin/sonar
status

SonarQube is not running.
Error: sonar4: failed
Failed executing: /usr/local/Cellar/sonar4/4.5.4/bin/sonar status
/usr/local/Library/Homebrew/formula.rb:730:in `block in system'
/usr/local/Library/Homebrew/formula.rb:687:in `open'
/usr/local/Library/Homebrew/formula.rb:687:in `system'
/usr/local/Library/Taps/homebrew/homebrew-versions/sonar4.rb:48:in `block in <class:Sonar4>'
/usr/local/Library/Homebrew/formula.rb:639:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:21:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:636:in `run_test'
/usr/local/Library/Homebrew/cmd/test.rb:38:in `block (2 levels) in test'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `block in test'
/usr/local/Library/Homebrew/cmd/test.rb:15:in `each'
/usr/local/Library/Homebrew/cmd/test.rb:15:in `test'
/usr/local/Library/brew.rb:135:in `<main>'

File: sonar4.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+class Sonar4 < Formula
+  homepage "http://www.sonarqube.org/"
+  url "http://dist.sonar.codehaus.org/sonarqube-4.5.4.zip"
+  sha1 "755d93b58d8fe88f4e7e99eb11930254128bc5c1"
+
+  def install
+    # Delete native bin directories for other systems
+    rm_rf Dir['bin/{aix,hpux,linux,solaris,windows}-*']
```
Comment (DomT4) [inline sonar4.rb]: Double quotes around text.

File: sonar4.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    libexec.install Dir['*']
+
+    if MacOS.prefer_64_bit?
+      bin.install_symlink "#{libexec}/bin/macosx-universal-64/sonar.sh" => "sonar"
+    else
+      bin.install_symlink "#{libexec}/bin/macosx-universal-32/sonar.sh" => "sonar"
+    end
+  end
+
+  plist_options :manual => "#{HOMEBREW_PREFIX}/opt/sonar4/bin/sonar console"
```
Comment (DomT4) [inline sonar4.rb]: Just make this ` plist_options :manual => "sonar console" `. Less trouble.

File: sonar4.rb (hunk 2 above)
Comment (DomT4) [inline sonar4.rb]:
```ruby
  test do
    assert_match /SonarQube/, pipe_output("#{bin}/sonar status")
  end
```

That'll work.

**Comment by a1dutch** [comment]
should be good now, I've squashed all the commits into one

**Comment by DomT4** [comment]
Merged! Thanks for your contribution to Homebrew, and squashing the commits into one :heart:. We appreciate it! For future reference the preferred commit message for new formula is ` sonar4 4.5.4 (new formula) `, Thanks!

**Comment by a1dutch** [comment]
Thanks for the help, i will update my pull request with the same changes to
the main sonar.


On 1 Mar 2015, at 18:55, Dominyk Tiller <notifications@github.com> wrote:

Merged! Thanks for your contribution to Homebrew, and squashing the commits
into one [image: :heart:]. We appreciate it! For future reference the
preferred commit message for new formula is sonar4 4.5.4 (new formula),
Thanks!

—
Reply to this email directly or view it on GitHub
<https://github.com/Homebrew/homebrew-versions/pull/692#issuecomment-76623918>
.

**Pull request closed without merging**